https://github.com/Toqozz/wired-notify/pull/185

From a797e34f01292ac4eaac647472f5704f6d8d9b0f Mon Sep 17 00:00:00 2001
From: classabbyamp <dev@placeviolette.net>
Date: Mon, 12 Oct 2026 13:18:29 -0400
Subject: [PATCH] manager: fix build on 32-bit architectures

without this cast, build fails with this error on 32-bit platforms,
because `XScreenSaverInfo.idle` has the type `c_ulong`, which is `u32`
on 32-bit platforms and `u64` on 64-bit platforms

---
 src/manager.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/manager.rs b/src/manager.rs
index 85e7504..15a1ed3 100644
--- a/src/manager.rs
+++ b/src/manager.rs
@@ -198,7 +198,7 @@ impl NotifyWindowManager {
                         // 1s to be considered idle.
                         let idle_last_frame = self.is_idle_1s();
                         let is_idle = info.idle / 1000 >= 1;
-                        let info_idle = info.idle;
+                        let info_idle = info.idle as u64;
 
                         // If we "woke up" this frame.
                         if cfg.unpause_on_input && idle_last_frame && !is_idle {
